### PR TITLE
Enhance/ia model building

### PR DESCRIPTION
**Pull request by Jordan-B1**

### CODE REVIEW

**Review by H4ss — APPROVED** [review]

File: ai/ai_model_building/army_builder.py [hunk 1]
```diff
@@ -2,13 +2,30 @@
...
 
 def army_builder(player_resume):
+    """Create a representation of an army from the player resume.
+        This representation is used as model input
+
+    Args:
+        player_resume (dict): The player army resume
+
+    Returns:
+        dict: The army representation
```
Comment (Pepiloto) [inline ai/ai_model_building/army_builder.py:27]: Why no indication about the types in the dict ?

File: ai/ai_model_building/army_builder.py [hunk 2]
```diff
@@ -19,14 +36,52 @@ def army_builder(player_resume):
...
 
 def match_builder(match_resume):
+    """Create a representation of a match from the match resume.
+     This representation is used as model input
+
+    Args:
+        match_resume (Object): The match resume
+
+    Returns:
+        dict: The match representation
```
Comment (Pepiloto) [inline ai/ai_model_building/army_builder.py:46]: Why no indication about the types in the dict ?

File: ai/ai_model_building/army_builder.py [hunk 3]
```diff
@@ -2,13 +2,30 @@
...
             return unit
     return None
 
 
 def army_builder(player_resume):
+    """Create a representation of an army from the player resume.
+        This representation is used as model input
+
+    Args:
+        player_resume (dict): The player army resume
```
Comment (Pepiloto) [inline ai/ai_model_building/army_builder.py:24]: Why no indication about the types in the dict ?

File: ai/ai_model_building/army_builder.py [hunk 4]
```diff
@@ -19,14 +36,52 @@ def army_builder(player_resume):
 
 
 def match_builder(match_resume):
+    """Create a representation of a match from the match resume.
+     This representation is used as model input
+
+    Args:
+        match_resume (Object): The match resume
```
Comment (Pepiloto) [inline ai/ai_model_building/army_builder.py:43]: What is Object ? If possible add a more significative typename

File: ai/ai_model_building/army_builder.py [hunk 5]
```diff
@@ -19,14 +36,52 @@ def army_builder(player_resume):
...
     second_player = army_builder(match_resume['second_player'])
     map = match_resume['map']
     return {'first_player': first_player, 'second_player': second_player, 'map': map}
 
 
 def build_trainning_data(matchs):
+    """This function take as parameter a list of matchs and return it build with the dataclass created so it can be used as model input
+
+    Args:
+        matchs (list): The list of matchs to build
```
Comment (Pepiloto) [inline ai/ai_model_building/army_builder.py:58]: Why no indication about the types in the list ?

File: ai/ai_model_building/army_builder.py [hunk 6]
```diff
@@ -19,14 +36,52 @@ def army_builder(player_resume):
...
+
+    Returns:
+        list: The list of matchs built
```
Comment (Pepiloto) [inline ai/ai_model_building/army_builder.py:61]: Why no indication about the types in the list ?

File: ai/ai_model_building/stat_parser.py [hunk 7]
```diff
@@ -33,6 +49,15 @@ def numerise_field(value):
...
 
 def load_unit_stat(stat, unit):
+    """Fetch the substats of a stat from a unit descriptor
+
+    Args:
+        stat (str): The stat's name
+        unit (str): The unit name
+
+    Returns:
+        dict: The substats of the stat
```
Comment (Pepiloto) [inline ai/ai_model_building/stat_parser.py:59]: Why no indication about the types in the dict ?

**Review by Pepiloto — CHANGES_REQUESTED** [review]

File: ai/ai_model_building/army_builder.py (hunk 1 above)
Comment (Jordan-B1) [reply ai/ai_model_building/army_builder.py:27]: That's problem I had while documenting this PR, the types are way too long but yeah I guess imma right it like that anyway :/

**Review by H4ss — APPROVED** [review]

File: ai/ai_model_building/README.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
+# PRophecy - AI model building
```
Comment (Pepiloto) [inline ai/ai_model_building/README.md:1]:
```suggestion
# Prophecy - AI model building
```

**Review by Pepiloto — DISMISSED** [review]

**Pull request merged**